src/dropdown/dropdown.ts

```diff
@@ -143,22 +135,27 @@ export class NgbDropdown implements OnInit, OnDestroy {
 
   constructor(
       private _changeDetector: ChangeDetectorRef, config: NgbDropdownConfig, @Inject(DOCUMENT) private _document: any,
-      private _ngZone: NgZone) {
+      private _ngZone: NgZone, private _elementRef: ElementRef<HTMLElement>, private _renderer: Renderer2) {
     this.placement = config.placement;
+    this.container = config.container;
     this.autoClose = config.autoClose;
     this._zoneSubscription = _ngZone.onStable.subscribe(() => { this._positionMenu(); });
   }
 
   ngOnInit() {
-    if (this._menu) {
-      this._menu.applyPlacement(Array.isArray(this.placement) ? (this.placement[0]) : this.placement as Placement);
-    }
+    this._setupMenuAttributes();
```

Comment:
Maybe better to do this on open only? what's the point of moving DOM around on widget initialization? If you have a table with a 100 of dropdowns it will do lots of useless operations

Reply:
I need to perform at least the applyPlacement, as it was done before, otherwise there are unit tests failing (i.e. ngb-dropdown-toggle Custom config should initialize inputs with provided config FAILED).

 I don't know if it worth to do it, but I'll add a condition to avoid the this._applyContainer(this.container); if the dropdown is not open.